openwisp_monitoring/device/tests/test_api.py

```diff
@@ -1009,3 +1012,247 @@ def test_bearer_authentication(self):
                 HTTP_AUTHORIZATION=f'Bearer {token}',
             )
             self.assertEqual(response.status_code, 200)
+
+
+class TestWifiClientSessionApi(
+    AuthenticationMixin, TestWifiClientSessionMixin, TestCase
+):
+    def _login_admin(self):
+        User = get_user_model()
+        admin = User.objects.create_superuser('admin', 'admin', 'test@test.com')
+        self.client.force_login(admin)
+
+    # Wifi sessions
+    def test_wifisession_unauthorized_api_access(self):
+        device = self._create_device()
+        wifi_session = self._create_wifi_session(device=device)
+        response = self.client.get(
+            reverse('monitoring:api_wifi_session_detail', args=(wifi_session.id,))
+        )
+        self.assertEqual(response.status_code, 401)
+        response = self.client.get(reverse('monitoring:api_wifi_session_list'))
+        self.assertEqual(response.status_code, 401)
+        response = self.client.patch(
+            reverse('monitoring:api_wifi_session_detail', args=(wifi_session.id,)),
+            data=json.dumps({'ssid': 'Free Public Wifi Updated'}),
+            content_type='application/json',
+        )
+        self.assertEqual(response.status_code, 401)
+        wifi_client = self._create_wifi_client(mac_address='22:33:44:55:66:88')
+        wifi_session_post_data = {
+            'device': str(device.pk),
+            'wifi_client': str(wifi_client.pk),
+            'ssid': 'Free Public WiFi',
+            'interface_name': 'wlan0',
+        }
+        response = self.client.post(
+            reverse('monitoring:api_wifi_session_list'),
+            data=json.dumps(wifi_session_post_data),
+            content_type='application/json',
+        )
+        # Unauthorized response
+        self.assertEqual(response.status_code, 401)
+
+    def test_wifisession_detail_get(self):
+        device = self._create_device()
+        wifi_session = self._create_wifi_session(device=device)
+        self._login_admin()
+        url = reverse('monitoring:api_wifi_session_detail', args=(wifi_session.id,))
+        response = self.client.get(url)
+        data = response.data
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(data), 11)
+        self.assertEqual(data['device_name'], 'default.test.device')
+        self.assertEqual(data['interface_name'], 'wlan0')
+        self.assertEqual(len(data['client']), 9)
+        self.assertEqual(data['client']['mac_address'], '22:33:44:55:66:77')
+        self.assertContains(response, 'created')
+        self.assertContains(response, 'modified')
+
+    def test_wifisession_detail_update(self):
+        device = self._create_device()
+        wifi_session = self._create_wifi_session(device=device)
+        self._login_admin()
+        url = reverse('monitoring:api_wifi_session_detail', args=(wifi_session.id,))
+        response = self.client.patch(
+            url,
+            data=json.dumps({'ssid': 'Free Public Wifi Updated'}),
```

Comment:
Tip: You don't need to `json.dumps` here. The test client should take care of it.